R/dv8.R

```diff
@@ -0,0 +1,153 @@
+                                '-outputFolder', flaws_path,
+                                hsdsmb_path),
+          stdout=FALSE, stderr=FALSE)
+}
+
+#' Parse Architecture Flaws
+#'
+#' @param file_measure_report_path path to file-measure-report.csv
+#' @param flaws_path path to architecture folder
+#' @export
+parse_dv8_architectural_flaws <- function(file_measure_report_path, flaws_path) {
+  folders_expected <- c("clique", "modularity-violation", "package-cycle", "unhealthy-inheritance")
+  folders_exist <- list.files(flaws_path)
+
+  check_folders_exist <- function(folders_expected, folder) {
+    folder_exists <- FALSE
+
+    if (folder %in% folders_expected) {
+      folder_exists <- TRUE
+    }
+
+    return(folder_exists)
+  }
+
+  convert_dsm_binary_to_json <- function(dsm_binary_file_path, json_path) {
+    hierjson_path <- file.path(json_path, 'hier.json')
+    system2('dv8-console', args=c('core:export-matrix', '-outputFile', hierjson_path, dsm_binary_file_path), stdout=FALSE, stderr=FALSE)
+    return (hierjson_path)
+  }
+
+  generate_file_paths <- function(folder_name, flaws_path) {
+    # Initialize an empty data.table to store the file paths
+    dt_files <- data.table::data.table(file_path = character(), architecture_issue_type = character(), architecture_issue_id = character())
+
+    # Get the full path to the current folder
+    folder_path <- file.path(flaws_path, folder_name)
+
+    # Check if the sub-folder exists
+    if (dir.exists(folder_path)) {
+      # Get a list of all sub-folders in the current folder
+      subdirs <- list.dirs(folder_path)
+      subdirs <- subdirs[-1]
+
+      # Loop over each sub-folders in the current folder
+      for (subdir in subdirs) {
+
+        # Get a list of all files in the current sub-folder
+        files <- list.files(subdir, full.names = TRUE)
+
+        # Get a list of all *merge.dv8-dsm file
+        files_merged <- grep('merge', files, value = TRUE)
+
+        # If there are any files that contain the word 'merge'
+        if (length(files_merged) > 0) {
+          # loop through each file path in files_merged
+          for (file_path in files_merged) {
+            # call convert_dsm_binary_to_json for each dv8-dsm file
+            save_path <- dirname(file_path)
+            result <- convert_dsm_binary_to_json(file_path, save_path)
+
+            # Get variables (which contains the file paths) from json file
+            file_paths <- jsonlite::fromJSON(result)
+            files <- file_paths$variables
+            file.remove(result)
+
+            for (project_file in files){
+              # add the file path and folder name to the data.table
+              issue_id <- gsub("[^0-9]", "", gsub(folder_path, '', subdir))
+              dt_files <- rbind(dt_files, data.table::data.table(file_path = project_file, architecture_issue_type = folder_name, architecture_issue_id = issue_id))
```

Comment:
Add a few line breaks here after the commas, e.g. 

file_path = project_file,
architecture_issue_type = folder_name, 

etc.